src/Cake.Common.Tests/Unit/Tools/MSBuild/MSBuildRunnerTests.cs

```diff
@@ -946,6 +946,51 @@ public void Should_Escape_Special_Characters_In_Property_Value(string propertyVa
                 // Then
                 Assert.Equal(expected, result.Args);
             }
+
+            [Fact]
+            public void Should_Use_WarnAsError_If_Specified()
+            {
+                // Given
+                var expected = "/v:normal /target:Build /warnaserror \"C:/Working/src/Solution.sln\"";
+                var fixture = new MSBuildRunnerFixture(false, PlatformFamily.Windows);
+                fixture.Settings.WithWarningsAsError();
+
+                // When
+                var result = fixture.Run();
+
+                // Then
+                Assert.Equal(expected, result.Args);
+            }
+
+            [Fact]
+            public void Should_Use_WarnAsError_Codes_If_Specified()
+            {
+                // Given
+                const string expected = "/v:normal /target:Build /warnaserror:\"12345\" \"C:/Working/src/Solution.sln\"";
+                var fixture = new MSBuildRunnerFixture(false, PlatformFamily.Windows);
+                fixture.Settings.WithWarningsAsError("12345");
+
+                // When
+                var result = fixture.Run();
+
+                // Then
+                Assert.Equal(expected, result.Args);
+            }
+
+            [Fact]
+            public void Should_Use_WarnAsMessage_Codes_If_Specified()
```

Comment:
@devlead do we need another test for the combination of both error and message?

Reply:
Do we? "separate" things, happy to add if you think it's needed though.

Reply:
Not saying that it is needed, but just asking the question, as it occurred to me as I was reading the code.  My thought process was that since these properties are intended to work together, a final overall test to show them working at the same time would be a good idea.  Perhaps overkill though.

Reply:
@gep13 added `Should_Use_WarnAsError_And_WarnAsMessage_Codes_If_Specified()`